### PR TITLE
STAR alignment wrapper containing 2 pass alignment and use of metadata sheet

### PR DESCRIPTION
**Pull request by akahles**
The code in the PR contains the wrapper implementing the 2 pass alignment approach that has been used to create the STAR alignments. Further it provides the possibility to align given the metadata sheet and an analysis ID instead of an xml file. Last, the PR updates the STAR version that is used from 2.4.0f1 to 2.4.0g1.

### CODE REVIEW

File: star_align.py [hunk 1]
```diff
@@ -70,6 +70,48 @@ def scan_workdir(base):
     raise Exception("Unable to determine input type")
 
 
+def spreadsheet2dict(spreadFile):
```
Comment (kellrott) [inline star_align.py]: This function is implemented in the Python standard library as cvs.DictReader ( https://docs.python.org/2/library/csv.html#csv.DictReader )

File: star_align.py [hunk 2]
```diff
@@ -116,8 +158,11 @@ def xml2RGdict(xmlfile):
     optional = parser.add_argument_group("optional input parameters")
     optional.add_argument("--out", default="out.bam", help="Name of the output BAM file")
     optional.add_argument("--workDir", default="./", help="Work directory")
+    optional.add_argument("--metaData", default=None, help="File containing metadata for the alignment header")
```
Comment (kellrott) [inline star_align.py]: Can we change this to --metaDataTab, so if we add other forms of metaData input later, we can break them out by input type?

Comment (akahles) [inline star_align.py]: Good point - I will follow your suggestion.

File: star_align.py (hunk 1 above)
Comment (akahles) [inline star_align.py]: My use case is a little different than the standard library. Whereas DictReader provides an iterator over rows, where I can address each record with the field names defined in the header row, my function returns a dictionary that is accessible by analysis ID. This way I can easily look up the record I need. Anyway, I will check if it is easier to use the reader to construct my dictionary.

Comment (kellrott) [inline star_align.py]: Ok, just a style comment, no need to make the change.

**Pull request merged**